docs/source/guides/optimizers.rst

```diff
@@ -0,0 +1,59 @@
+Optimizer
+~~~~~~~~~
+
+.. include:: ../imports.rst
+
+In order to get good values for parameters, we have to optimize them by the :class:`Optimizer` class.
+It runs a numerical optimization algorithm on a given link.
+Many algorithms are implemented in the :mod:`~chainer.optimizers` module.
+Here we use the simplest one, called Stochastic Gradient Descent (SGD):
+
+.. doctest::
+
+   >>> model = MyChain()
+   >>> optimizer = optimizers.SGD()
+   >>> optimizer.setup(model)
+
+The method :meth:`~Optimizer.setup` prepares for the optimization given a link.
+
+Some parameter/gradient manipulations, e.g. weight decay and gradient clipping, can be done by setting *hook functions* to the optimizer.
+Hook functions are called after the gradient computation and right before the actual update of parameters.
+For example, we can set weight decay regularization by running the next line beforehand:
+
+   >>> optimizer.add_hook(chainer.optimizer.WeightDecay(0.0005))
+
+Of course, you can write your own hook functions.
+It should be a function or a callable object, taking the optimizer as the argument.
```

Comment:
Oh, I don't think the argument should be the optimizer. https://github.com/chainer/chainer/blob/master/chainer/optimizer.py#L777

Reply:
This is done in several of our examples, c.f. [dcgan](https://github.com/chainer/chainer/blob/9b2fcbe89a032f82af0d3dd8023e7301c9560eed/examples/dcgan/train_dcgan.py), [cifar](https://github.com/chainer/chainer/blob/89f8e1e05337a4f07096f6696d512346bf60fc02/examples/cifar/train_cifar_custom_loop.py), [post tagging](https://github.com/chainer/chainer/blob/0080f6a3f2036f03d50b661a23c2558cd7c50974/examples/pos/postagging.py).

Reply:
Hm, I think this should fixed.